src/libraries/System.Data.Common/tests/TrimmingTests/DbConnectionStringBuilder.cs

```diff
@@ -0,0 +1,96 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Data.Common;
+using System.ComponentModel;
+using System.Collections;
+
+class Program
+{
+    static int Main(string[] args)
+    {
+        DbConnectionStringBuilder2 dcsb2 = new();
+        ICustomTypeDescriptor td = dcsb2;
+        if (td.GetClassName() != "DbConnectionStringBuilder2")
+        {
+            throw new Exception("Class name got trimmed");
+        }
+
+        if (td.GetComponentName() != "Test Component Name")
+        {
+            throw new Exception("Component name got trimmed");
+        }
+
+        bool foundAttr = false;
+        string attrName = "Test" + ((args != null) ? "Attribute" : "");
```

Comment:
Why are we checking if `args != null` here?

Reply:
I wasn't sure if linker searches for patterns like "attr.GetType().Name == CompileTimeConstant" so I made it a dummy expression which always evaluates to "TestAttribute" but made it hard for analyzers to detect...

Reply:
Linker doesn't handle attributes in any special way currently. Definitely not patterns like this.

Reply:
will simplify then - I didn't want the test to pass by accident :)